AzureDevOpsPipelines/deployment/bot-publish/source/Core.ts

```diff
@@ -0,0 +1,110 @@
+import tl = require("azure-pipelines-task-lib");
+import { IExecSyncResult, IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import { execSync } from "child_process";
+import path = require("path");
+import fs = require("fs");
+
+export class core {
+    private isLoggedIn: boolean = false;
+    private cliPasswordPath: string = "";
+    private servicePrincipalId: string = "";
+    private servicePrincipalKey: string = "";
+    private prefix: string = "";
+    
+    public subscriptionID: string = "";
+
+    constructor() {
+        this.getPrefix();
+        this.subscriptionID = tl.getVariable('subscriptionID');
+        
+        if (this.subscriptionID == null) {
+            this.loginAzureRM(tl.getInput("connectedServiceNameARM", true));
+            tl.setVariable('subscriptionID', this.subscriptionID);
+        }
+    }
+
+    private loginAzureRM(connectedService: string): void {
+        var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, true);
```

Comment:
add a space below this line. Also, it appears that this variable is only set here, so it'd be a const